test/al_util_test.js

```diff
@@ -0,0 +1,42 @@
+const assert = require('assert');
+const rewire = require('rewire');
+const sinon = require('sinon');
+const rp = require('request-promise-native');
+const req = require('request');
+const RestServiceClient = require('../al_util').RestServiceClient;
+const m_alMock = require('./al_mock');
+const debug = require('debug') ('azcollectc_test');
+var servicecRewire = rewire('../al_servicec');
+var m_servicec = require('../al_servicec');
+
+describe('Unit Tests', function() {
+
+    describe('RestServiceClient', function() {
+        it('check _initRequestOptions', function(done) {
```

Comment:
@kkuzmin 
Ideally, I'd like to test every method of `RestServiceClient` here, but for that we need to make `rp`. Do you have any ideas on how to do it correctly?

I tried this:

```js

const req = require('request');


var fakeRp;

beforeEach(function() {
    fakeRp = sinon.stub(req, 'request').callsFake(
        function fakeFn(newOptions) {
            return new Promise(function(resolve, reject) {
                resolve({response : 'some_response'});
            });
        }
    );
});

afterEach(function() {
    fakeRp.restore();
});


it('check request', function(done) {
    var restC = new RestServiceClient(m_alMock.AL_API);
    var body = {
        field1 : true,
        field2: 'false'
    };
    restC.request('POST', '/some/path', { body : body })
    .then(resp => {
        sinon.assert.calledWith(fakeRp,
            {
                method: 'POST',
                url: 'https://al-api-endpoint.alertlogic.com/some/path',
                json: true,
                headers: {
                    'Accept': 'application/json'
                },
                pool: {
                    maxSockets: 128
                },
                body : body
            }
        );
        done();
    });
});
```
but it fails 😞 

```
(node:28575) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): RequestError: Error: getaddrinfo ENOTFOUND al-api-endpoint.alertlogic.com al-api-endpoint.alertlogic.com:443
(node:28575) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```